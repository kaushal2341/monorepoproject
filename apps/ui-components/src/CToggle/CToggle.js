import React, {memo} from 'react';
import BootstrapSwitchButton from 'bootstrap-switch-button-react';
import PropTypes from 'prop-types';

const CToggle = props => {
    const {
        id, onLabel, offLabel,
        size, onStyle, offStyle,
        width, height, name,
        onChangeHandler, classes, required,
        checked, style
    } = props;
    return (
        <>
            <BootstrapSwitchButton
                id={id}
                onlabel={onLabel}
                offlabel={offLabel}
                size={size}
                onstyle={onStyle}
                offstyle={offStyle}
                style={style}
                width={width}
                height={height}
                className={classes}
                onChange={onChangeHandler}
                name={name}
                checked={checked}
                required={required}
            />
        </>
    )
}

export default memo(CToggle);

CToggle.defaultProps = {
    id: "c-toggle",
    name: "c-toggle",
    onChangeHandler: () => {
    },
    checked: false,
}

CToggle.propTypes = {
    id: PropTypes.string || PropTypes.number,
    name: PropTypes.string.isRequired,
    onLabel: PropTypes.string || PropTypes.element,
    offLabel: PropTypes.string || PropTypes.element,
    size: PropTypes.number,
    onStyle: PropTypes.string,
    offStyle: PropTypes.string,
    style: PropTypes.string,
    className: PropTypes.string,
    onChange: PropTypes.func.isRequired,
    checked: PropTypes.bool.isRequired,
    width: PropTypes.number,
    height: PropTypes.number
}
