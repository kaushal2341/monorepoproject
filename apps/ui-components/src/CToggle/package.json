{
  "name": "@cogent/c-toggle",
  "version": "0.0.0",
  "main": "CToggle.js",
  "license": "MIT",
  "private": true
}