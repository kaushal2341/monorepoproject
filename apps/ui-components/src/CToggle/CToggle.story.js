import React from 'react';
import {storiesOf} from "@storybook/react";

import CToggle from './CToggle';

storiesOf('CToggle', module)
    .add('default', () => <CToggle onLabel="On" offLabel="Off"/>);