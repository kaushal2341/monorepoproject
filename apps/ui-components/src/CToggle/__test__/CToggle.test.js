import CToggle from '../CToggle';

expect.addSnapshotSerializer(enzymeSerializer);
describe('CToggle Component', () => {
    let wrapper, children, mock;
    beforeEach(() => {
        mock = jest.fn();
        wrapper = shallow(<CToggle id="c-toggle" onChangeHandler={mock}/>);
        children = wrapper.find('#c-toggle');
    });

    it('should be defined', () => {
        expect(wrapper).toBeDefined();
    });

    it('should have only one toggle button', () => {
        expect(children).toHaveLength(1);
    });

    it('should have given props available', () => {
        const propsForToggle = ['id', 'name', 'className',
            'onlabel', 'offlabel', 'size', 'onstyle',
            'offstyle', 'width', 'height', 'style',
            'onChange', 'required', 'checked'];
        const propsForToggleAvailable = Object.keys(children.props());
        propsForToggle.map((propToggle, index) => {
            expect(propsForToggle).toContain(propsForToggleAvailable[index]);
        });
    });

    it('should provide default props when the props is not available', () => {
        expect(children.prop('checked')).toBeFalsy();
        expect(children.prop('id')).toBe('c-toggle');
        expect(children.prop('name')).toBe('c-toggle');
        expect(typeof children.prop('onChange')).toBe('function');
    });

    it('should provide the props to the component when props is given', () => {
        wrapper.setProps({
            checked: true,
            id: 'c-toggle1',
            name: 'c-toggle1',
            onChange: jest.fn()
        });
        const children1 = wrapper.find('#c-toggle1');
        expect(children1.prop('checked')).toBeTruthy();
        expect(children1.prop('id')).toBe('c-toggle1');
        expect(children1.prop('name')).toBe('c-toggle1');
        expect(typeof children1.prop('onChange')).toBe('function');
    });

    it('should simulate change when the switch button is change', () => {
        children.simulate('change', {target: {name: 'c-toggle', value: true}});
        expect(mock).toHaveBeenCalled();
    });

    describe('Snapshot Testing', () => {
        it('should match previous snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});