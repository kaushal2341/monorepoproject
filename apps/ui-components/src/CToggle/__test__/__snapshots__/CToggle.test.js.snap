// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CToggle Component Snapshot Testing should match previous snapshot 1`] = `
<Fragment>
  <n
    checked={false}
    id="c-toggle"
    name="c-toggle"
    onChange={[MockFunction]}
  />
</Fragment>
`;
